### ISSUE TITLE
Web: Finance Suite (Host Payouts, Guest Payments, Accounting) main

### ISSUE THREAD

**Issue by Timmhart** [domits1/Domits#163]
## 🚀 Feature Summary

The Finance Suite feature centralizes all financial operations across bookings and properties. It tracks payouts, commissions, fees, and taxes with clear reconciliation and reporting. Users gain real-time financial visibility, accuracy, and control to support accounting and business decisions.

---

[PMS](https://github.com/domits1/Domits/issues/149) Capabilities: [Listings](https://github.com/domits1/Domits/issues/2792), [CRS](https://github.com/domits1/Domits/issues/501), [CM](https://github.com/domits1/Domits/issues/440), [Direct Bookings](https://github.com/domits1/Domits/issues/2600), [RMS](https://github.com/domits1/Domits/issues/436), [Calendar & Prices](https://github.com/domits1/Domits/issues/36), [Inbox](https://github.com/domits1/Domits/issues/767), [Tasks](https://github.com/domits1/Domits/issues/2794), [Finance](https://github.com/domits1/Domits/issues/163), [Reviews](https://github.com/domits1/Domits/issues/90), [Intelligence](https://github.com/domits1/Domits/issues/1490), [Settings](https://github.com/domits1/Domits/issues/2793), [Integrations](https://github.com/domits1/Domits/issues/433), [Agentic AI](https://github.com/domits1/Domits/issues/442)
Use Case / Scenario Testing: [PMS](https://github.com/domits1/Domits/issues/2262), [CRS](https://github.com/domits1/Domits/issues/2265), [CM](https://github.com/domits1/Domits/issues/2266), [RMS](https://github.com/domits1/Domits/issues/2267), [Calendar & Prices](https://github.com/domits1/Domits/issues/2299), [Inbox](https://github.com/domits1/Domits/issues/2291), [Finance](https://github.com/domits1/Domits/issues/2287)

----

<img width="1306" height="517" alt="Image" src="https://github.com/user-attachments/assets/7e352243-13df-4bf1-b51a-927300d8d653" />

----

# Main issue

**System**
- [ ] #2256

**Host Payouts**
- [x] #175 
- [x] #1664
- [x] #1910
- [ ] #1665
- [ ] #2258
- [ ] #2623
- [ ] Host PMS Dashboard Finance: Payouts: Currency conversion
- [ ] #2259
- [x] Finance FAQ at the bottom
- [ ] Add a guide for payouts (timing/method), documents & invoices, financial overview, finance settings. https://www.domits.com/helpdesk-host
- [ ] #2985 

**Guest Payments**
- [ ] #145
- [x] #622
- [x] #1911
- [ ] #623
- [ ] #2369

**Accounting**
- [ ] #984
- [ ] #2597
- [ ] Automated bank sync reconciliation
- [ ] Automated owner statements

**Financial Settings**
- [ ] #3112

**Future Finance APIs for integrations**
- [ ] https://github.com/domits1/Domits/issues/2295
- [x] #2297 
- [ ] #2927 
- [ ] Finance API: Full access to invoices, folios and all accounting operations.

---

## 🧩 Problem Statement
What problem does this feature solve? Who is affected, and why does it matter? > Focus on the problem, not the solution.

No centralized finance suite

---

## 🎯 Desired Outcome
What does success look like if this feature is implemented?

Centralized finance suite

---

## 👥 Target Users
Who benefits from this feature?
- [x] Guests / Customers
- [x] Hosts / Property Managers / Enterprise
- [ ] Admin / Internal users
- [ ] Developers
- [ ] Other: _______

---

## 📈 Business & User Impact
How valuable is this feature?

- **Impact level:**
  - [ ] Low (nice-to-have)
  - [ ] Medium (improves workflows)
  - [ ] High (unblocks key use cases)
  - [x] Critical (strategic / revenue-impacting)

- **User reach:**
  - [ ] Few users
  - [ ] Many users
  - [x] All users
  - [ ] Unknown

---

## ⚠️ Constraints & Considerations
- Performance
- Security
- Privacy
- Accessibility
- Backward compatibility
- External dependencies

---

## 🧪 Acceptance Criteria / User Stories
How can we validate that this feature is complete?

👥 User Roles

- Guest
- Host / Property Owner
- Property Manager / Operator
- Finance Manager / Accountant
- Admin / System Operator
- Integration System (Payment Gateway, Accounting API, Bank, OTA)

💳 1. Payments & Transactions

- [x] As a guest, I want to pay securely online via credit card, PayPal, or bank transfer.
- [x] As a guest, I want to see a detailed cost breakdown before confirming my booking.
- [ ] As a system, I want to automatically charge the guest based on booking rules (deposit, balance).
- [ ] As a host, I want to set up automated payment schedules (e.g., 50% at booking, 50% before check-in).
- [ ] As a host, I want to accept multiple currencies.
- [ ] As a finance manager, I want to record all incoming payments by date and property.
- [ ] As a system, I want to support partial payments.
- [ ] As a system, I want to store payment references securely for auditing.
- [ ] As an admin, I want to enable or disable specific payment methods per property.
- [ ] As a host, I want to collect security deposits automatically.

🧾 2. Invoicing & Receipts

- [ ] As a host, I want invoices to be automatically generated upon booking confirmation.
- [ ] As a guest, I want to download my invoice after payment.
- [ ] As a host, I want to customize invoice templates with my logo and address.
- [ ] As a finance manager, I want to batch-generate invoices for multiple bookings.
- [ ] As a system, I want to number invoices sequentially and uniquely.
- [ ] As an admin, I want to define tax rules and automatically apply them to invoices.
- [ ] As a host, I want to resend invoices via email.
- [ ] As a finance manager, I want to generate credit notes for canceled or refunded bookings.
- [ ] As a guest, I want to receive electronic receipts automatically.
- [ ] As an admin, I want to apply digital signatures or seals on invoices for compliance.

💰 3. Refunds & Adjustments

- [ ] As a guest, I want to receive automatic refunds for cancellations within policy.
- [ ] As a host, I want to manually issue refunds for exceptional cases.
- [ ] As a system, I want to calculate refund amounts based on cancellation timing.
- [ ] As a finance manager, I want to track pending refunds for cash flow planning.
- [ ] As a system, I want to prevent double refunds for the same booking.
- [ ] As a host, I want to apply partial refunds for early departures.
- [ ] As a finance manager, I want to generate refund reports by property or channel.
- [ ] As a system, I want to update OTA channels when refunds are processed.

🏦 4. Payouts to Owners & Partners

- [x] As a host, I want to receive payouts automatically after guest check-in or check-out.
- [x] As a property manager, I want to define payout schedules (weekly, monthly, after stay).
- [x] As a system, I want to deduct management commissions before payout.
- [ ] As a finance manager, I want to view payout statements per owner.
- [ ] As a system, I want to handle multiple bank accounts for different owners.
- [ ] As an admin, I want to manage payout batch processing.
- [ ] As a finance manager, I want to download payout history in Excel format.
- [ ] As a system, I want to automatically hold payouts for disputed bookings.
- [ ] As a property manager, I want to split payouts between owners and cleaners if required.
- [ ] As an admin, I want to mark payouts as “pending,” “completed,” or “on hold.”

💼 5. Owner Statements & Accounting Reports

- [ ] As an owner, I want to view monthly financial statements for my property.
- [ ] As a manager, I want to see detailed transaction breakdowns by booking.
- [ ] As a finance manager, I want to generate profit and loss reports by property.
- [ ] As an owner, I want a clear summary showing gross revenue, deductions, and net payout.
- [ ] As a finance manager, I want to export financial reports to Excel or CSV.
- [ ] As a system, I want to automatically group income and expenses by account type.
- [ ] As a finance manager, I want to view historical performance trends.
- [ ] As a property manager, I want to generate consolidated reports for multiple properties.
- [ ] As an owner, I want automated monthly statements emailed to me.
- [ ] As an admin, I want to adjust report periods (weekly, monthly, quarterly).

🧮 6. Expense Tracking & Cost Management

- [ ] As a host, I want to record expenses like cleaning, utilities, and maintenance.
- [ ] As a system, I want to categorize expenses automatically based on type.
- [ ] As a finance manager, I want to attach receipts or invoices to expense records.
- [ ] As a system, I want to allocate expenses to specific properties.
- [ ] As a host, I want to set recurring expenses (e.g., monthly Wi-Fi bill).
- [ ] As a finance manager, I want to view expense-to-revenue ratios.
- [ ] As a system, I want to calculate net income after expenses.
- [ ] As an admin, I want to approve or reject submitted expenses.
- [ ] As a host, I want to split shared expenses across multiple properties.
- [ ] As a finance manager, I want to view expense trends over time.

📈 7. Taxes & Compliance

- [ ] As an admin, I want to define tax rates (VAT, occupancy, city taxes) per region.
- [ ] As a system, I want to apply local taxes automatically to bookings.
- [ ] As a finance manager, I want to generate tax summary reports by jurisdiction.
- [ ] As a system, I want to distinguish between inclusive and exclusive tax pricing.
- [ ] As a host, I want to automatically remit occupancy tax to local authorities.
- [ ] As an admin, I want to support multiple tax rules for multi-location properties.
- [ ] As a finance manager, I want to view tax liabilities by period.
- [ ] As a system, I want to produce exportable tax files for accountants.
- [ ] As an admin, I want to configure withholding taxes for owner payouts.
- [ ] As a system, I want to comply with government e-invoicing regulations.

🧾 8. Commission Management

- [ ] As an admin, I want to define commission rates per property or contract.
- [x] As a system, I want to deduct platform commissions automatically from gross revenue.
- [ ] As a host, I want to view total commissions charged each month.
- [ ] As a finance manager, I want to adjust commission rules for specific channels.
- [ ] As a system, I want to calculate manager commissions based on net payout.
- [ ] As an admin, I want to run a report of all commissions collected.
- [ ] As a system, I want to handle tiered commissions (e.g., lower rate for high volume).
- [ ] As a finance manager, I want to override commissions manually in special cases.
- [ ] As an admin, I want to apply discount codes or commission waivers temporarily.
- [ ] As a system, I want to display commission deductions on owner statements.

🔁 9. Channel & Payment Gateway Integration

- [ ] As a system, I want to receive OTA payment data and sync it automatically.
- [ ] As an admin, I want to manage API keys for each payment processor.
- [ ] As a system, I want to verify transaction success via API before marking a booking as paid.
- [ ] As an admin, I want to reconcile OTA financial data with PMS records.
- [ ] As a system, I want to import payout data from Airbnb or Booking.com automatically.
- [ ] As a finance manager, I want to map payment and fee structures across channels.
- [ ] As an admin, I want to integrate with Stripe, PayPal, or Adyen for direct payments.
- [ ] As a system, I want to automatically handle currency conversions from OTAs.
- [ ] As an admin, I want to configure default payment gateway per region.
- [ ] As a system, I want to retry failed transactions automatically.

🧠 10. Automation & Smart Finance Rules

- [ ] As a host, I want automated billing reminders for unpaid balances.
- [ ] As a system, I want to apply discounts automatically for long stays.
- [ ] As a system, I want to trigger payouts when all guests in a booking have checked out.
- [ ] As a system, I want to reconcile all transactions nightly.
- [ ] As an admin, I want to set up alerts for failed or delayed payouts.
- [ ] As a finance manager, I want automatic expense-to-income summaries sent monthly.
- [ ] As a system, I want to lock financial records after the month is closed.
- [ ] As a host, I want auto-notifications when funds are received.
- [ ] As a system, I want to auto-categorize income (rent, cleaning fees, extra charges).
- [ ] As a system, I want to detect and flag duplicate transactions.

🔐 11. Security, Audit & Compliance

- [ ] As a system, I want to encrypt all financial transactions.
- [ ] As an admin, I want to restrict finance access to authorized users only.
- [ ] As an auditor, I want to see all transaction history and changes.
- [ ] As a system, I want to maintain audit trails for edits and refunds.
- [ ] As an admin, I want role-based permissions for invoices, payouts, and expenses.
- [ ] As a system, I want to comply with PCI-DSS for card processing.
- [ ] As a finance manager, I want to export ledger data for external audits.
- [ ] As an admin, I want to anonymize old financial data after retention periods.

Later
- [ ] Add Bank details
Bank details Domits.com (part of Financieelsysteem B.V.)
ING Bank
Kinderhuissingel 6k
NL-2013 AS Haarlem (Amsterdam Area)
SWIFT: INGBNL2AXXX
Account Holder: Financieelsysteem B.V. 
IBAN: NL96INGB0105984663

**Comment by MC38-03** [comment]
Afgelopen periode:
- Stripe connect account creatie van front-end naar back-end verplaatst, dat houd in:
Om accounts te linken aan de main account voor Domits is er een Secret Key nodig vanuit Stripe Connect. Deze zette ik normaal gesproken in een .env file maar besloot uiteindelijk dat dat niet veilig was.

Nu zit deze Secret Key veilig in de environment variables van de Lambda function genaamd CreateStripeAccount.

Om Stripe te laten werken in deze Lambda function, heb ik een Layer moeten gebruiken. Dat houdt in een zip file waarin de node.js files zitten en de bijbehorende Stripe package die je dan importeerd als Layer. 

De code in de Lambda function maakt een onboarding link aan voor de Connect account met de Email van de persoon die het gaat maken. Deze link wordt vervolgens via de HostOnboardingAPI weer gestuurd naar de front-end, waar de user wordt gestuurd naar de link.

Vervolgens de code in de lambda function aangepast zodat deze account_id's van de aangemaakte Stripe Connect accounts worden opgeslagen in DynamoDB. 

Daarvoor was weer een IAM policy nodig, die toestemming gaf aan de Lambda function om records te zetten in DynaomoDB.

Daarbuiten is er een tweede Lambda function die ook een layer heeft voor Stripe. Deze maakt de betalingen zelf aan. Deze heeft ook zijn eigen API en connectie met de front-end. De test versie werkt door een vooraf ingesteld account_id te gebruiken. Maar dat gaan we natuurlijk dynamisch maken.

Komende periode:

- Stripe onboarding regelen voor users die eerst normaal hebben geregistreerd en daarna hebben gekozen om host te worden.

- De API endpoints zijn op dit moment in de front-end code, er moet een manier gevonden worden zodat we de endpoints niet in de front-end exposen maar dat ze vanuit AWS worden gekozen.

- Ervoor zorgen dat de informatie over het product in de checkout dynamisch wordt. (De accommodaties horen dus met deze informatie in een database te zitten en daar uit gehaald te worden wanneer een checkout wordt gemaakt ervoor.)

**Comment by MC38-03** [comment]
- [ ] Host Stripe Connect onboarding
- [ ] Accommodation info dynamically taken into checkout
- [ ] Checkout successfully working
- [ ] Guest payments tab
- [ ] Guest payments tab with more information
- [ ] Host payments tab
- [ ] Refund possibility

**Comment by TimSeijlhouwer** [comment]
-Van testmodus naar live modus
-Bedrag met komma naar juiste bedrag aanpassen(Chahid issue maken rondom dat het bedrag fout wordt meegenomen)
-Nieuwe designs binnen dynamic payment conformation and decline page
-Kijken of gevoelige informatie niet in de console terecht komt of ergens anders
-
